conda/recipes/ast_canopy/meta.yaml

```diff
@@ -20,29 +20,23 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}
+    - {{ stdlib('c') }}
     - cmake >=3.28
-    - pybind11
-    - clangdev >=18
     - ninja
-    - sysroot_{{ target_platform }} 2.17
   host:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - cuda-version {{ cuda_compiler_version }}
     - python
+    - pybind11
     - pip
     - scikit-build-core
+    - clangdev >=18
     - llvmdev >=14
     - llvm >=14
   run:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
+    - cuda-version >=12.5
+    - cuda-nvcc >=12.5
+    - cuda-nvcc_{{ target_platform }} >=12.5
+    - gcc >=10,<13
```

Comment:
```suggestion
```

This should be captured in the cuda-nvcc metapackage dependencies, unless this constraint is specific to ast_canopy and not cuda-nvcc.

Reply:
Dropping this constraint caused the solver to do some strange things like using gcc 7.

There is a constraint on cuda-nvcc-impl: https://github.com/conda-forge/cuda-nvcc-impl-feedstock/blob/938b371b79b824c439485ac52b29553444d31835/recipe/meta.yaml#L56

However, while nvcc supports something like gcc >=6,<14.0a0, I really don’t think gcc 7 is a viable solution, hence why I added this line. It suggests to me that there is some issue in compiler packages that causes the solver to fall back to a very old version.

Reply:
Is `gcc_impl_*` the same as `gcc`? From the feedstock `gcc` also provide symlinks to the compiler in addition to `gcc_impl_*`: https://github.com/conda-forge/ctng-compilers-feedstock/blob/main/recipe/meta.yaml. The question is which one does `cuda-nvcc` really needs, and should this be burdened to downstream packages?

Reply:
In most cases, they are the same. The `_impl_*` is an abstraction that facilitates cross-compiling. The constraints that `cuda-nvcc` has on the actual `gcc_impl_*` may not be correct, if cuda-nvcc might theoretically be used for cross-compiling. The `gcc_impl_*` dependencies are more appropriate in `cuda-nvcc_linux-64`. If NVCC always needs GCC of some sort, then a plain `gcc` constraint in `cuda-nvcc` would be appropriate.